### PR TITLE
Quirks Reworked 2: Autolathes aren't broken this time I swear boogaloo

### PR DESCRIPTION
**Pull request by bardinatavern4**
Hillarious, comedic even, funny if you will.

Autolathes now have a variable called complex. If true, it will check if you have technophobe and prevent you from using the autolathe if it's true.

legionarres and tribals have had the technophobe trait swapped for a new trait called Tribal. The code has been changed so bitter drink/healing powder now check for tribal and not technophobe.

An extra var has been added called mob_extratrait so quirks can have more then 1 trait, used only right now for tribal but may be used in the future.

### CODE REVIEW

**Comment by bardinatavern4** [comment]
Fixing the repo merge issues in a minute, I'm running an Only War Game

**Comment by AdamElTablawy** [comment]
Did you test this this time

**Comment by Lobachevskiy** [comment]
Why not just store the required trait in the variable instead of vaguely named "complex" var that you need to search references to even understand. Seriously, if you need to explain what the variable does in the description of a PR, it probably needs to be reworked. The added benefit is that you can then change the required trait in subclasses in the future.

Also you don't need to compare boolean variables with TRUE or FALSE, they already are one of those values. if(boolean) works.

File: code/datums/traits/_quirk.dm [hunk 1]
```diff
@@ -12,6 +12,7 @@
 	var/antag_removal_text // Text will be given to the quirk holder if they get an antag that has it blacklisted.
 	var/mood_quirk = FALSE //if true, this quirk affects mood and is unavailable if moodlets are disabled
 	var/mob_trait //if applicable, apply and remove this mob trait
+	var/mob_extratrait //for the occasional quirk that adds more then one trait
```
Comment (Lobachevskiy) [inline code/datums/traits/_quirk.dm:15]: Could just be a list to avoid copy paste

**Comment by bardinatavern4** [comment]
> Why not just store the trait blocking you from using the lathe in the variable instead of vaguely named "complex" var that you need to search references to even understand. Seriously, if you need to explain what the variable does in the description of a PR, it probably needs to be reworked. The added benefit is that you can then change the required trait in subclasses in the future.
> 
> Also you don't need to compare boolean variables with TRUE or FALSE, they already are one of those values. if(boolean) works.

The logic for this is so that it'll be set by default for all lathes to be unusuable by those with the technophobe trait. You want to have the option to set each lathe subtype (such as ammo benches) as being usable by technophobes so they aren't completely crippled. By adding a variable that all lathes have set to true by default, I can then go to a lathe I want to work for technophobes and set it to false to let them use that specific lathe. I did rename the variable to "tooadvanced" and put a comment by it explaining what it does.

File: code/datums/traits/_quirk.dm (hunk 1 above)
Comment (bardinatavern4) [reply code/datums/traits/_quirk.dm]: So just using mob_trait = list(the list) didn't work. I ended up just making it so the quirk has a datum that just adds the two traits. Bit blunt, but functional and doesn't mess with quirk.dm at all.

**Comment by bardinatavern4** [comment]
> Did you test this this time

Love you too Adam. 

Testing included:
Testing Longpork Stew with and without the cannibal trait, confirming that it hurt non-cannibals and slightly healed cannibals as well as metabolized faster when not hurt. Made sure longpork stew tastes good to cannibals but grosses out non-cannibals, is craftable, and that the liquid is colored when removed from the bowl.

Making sure the former tribal quirk gives you the correct traits, that technophobe still lets you use the reloading bench, that it does NOT let you use autolathes, and that non-technophobes still can use all kinds of bench/lathe

Making sure that the Legion/Tribals still got improved healing from Bitter Root and Healing Powder from the tribal trait, which they did.

Gonna add some more stuff but thats most of the testing for now

**Pull request merged**